Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java

```diff
@@ -0,0 +1,42 @@
+package mage.abilities.common;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+
+/**
+ *
+ * @author weirddan455
+ */
+public class DiceRolledTriggeredAbility extends TriggeredAbilityImpl {
+
+    public DiceRolledTriggeredAbility(Effect effect) {
+        super(Zone.BATTLEFIELD, effect);
+    }
+
+    private DiceRolledTriggeredAbility(final DiceRolledTriggeredAbility effect) {
+        super(effect);
+    }
+
+    @Override
+    public DiceRolledTriggeredAbility copy() {
+        return new DiceRolledTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DICE_ROLLED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return event.getPlayerId().equals(getControllerId());
```

Comment:
All other cards with `DICE_ROLLED` trigger uses additional `event.getFlag()` here (it's a custom param that can be used in events, see example in param's comment). I can't find the real usage of that flag in `ROLL_DICE` replace events, but it must be used same way here or deleted from other DICE_ROLLED's cards as unused.

![shot_210703_000201](https://user-images.githubusercontent.com/8344157/124323444-6324db00-db92-11eb-9ce2-ffe909d9e03c.png)


C:\Users\JayDi\Pictures\CloudShot\shot_210703_000201.png

Reply:
we're gonna have to do some refactoring anyway like I did with coin flips, so we'll want a `RollDiceEvent` for replacement effects and a `DiceRolledEvent` class for triggers

Reply:
@weirddan455 if you want to take this assignment then feel free

Reply:
Well I guess I'm not sure what data we'll need to store in those classes.  I just saw [[Barbarian Class]] has been spoiled which looks like we'll at least need to support multiple dice roll in a replacement effect and I'm not sure what else is coming.

Reply:
yeah, that’s the fun of preview season! luckily the set will be fully previewed by wednesday except for the commander decks

Reply:
@weirddan455 multi rolls already supported in replace event. See [[The Big Idea]] as example.

Reply:
I see.  Why do we need to refactor into new classes then @theelk801 ?

Reply:
because I didn't look too closely into it and assumed we had to

Reply:
Ok, I'm not saying it's a bad idea.  Was just curious.  I haven't looked into it too much either.  Maybe it's best to wait until full set gets spoiled so we know what all needs to be done.

Reply:
yeah that's basically what I was planning to do